### PR TITLE
[SYCL][USM] Implement depends_on inside command group handler.

### PR DESCRIPTION
**Pull request by jbrodman**
Adds ability to specify event or events that are required to complete before a command group may execute.

Signed-off-by: James Brodman <james.brodman@intel.com>

### CODE REVIEW

File: sycl/include/CL/sycl/detail/cg.hpp [hunk 1]
```diff
@@ -370,11 +376,13 @@ class CGExecKernel : public CG {
...
                std::vector<std::shared_ptr<const void>> SharedPtrStorage,
                std::vector<Requirement *> Requirements,
+               std::vector<detail::EventImplPtr> Events,
                std::vector<ArgDesc> Args, std::string KernelName,
                detail::OSModuleHandle OSModuleHandle,
                std::vector<std::shared_ptr<detail::stream_impl>> Streams)
       : CG(KERNEL, std::move(ArgsStorage), std::move(AccStorage),
-           std::move(SharedPtrStorage), std::move(Requirements)),
+           std::move(SharedPtrStorage), std::move(Requirements),
+           std::move(Events)),
```
Comment (romanovvlad) [inline sycl/include/CL/sycl/detail/cg.hpp]: It seems this should be clang formatted.

File: sycl/source/detail/scheduler/graph_builder.cpp [hunk 2]
```diff
@@ -483,6 +483,7 @@ Command *
 Scheduler::GraphBuilder::addCG(std::unique_ptr<detail::CG> CommandGroup,
                                QueueImplPtr Queue) {
   std::vector<Requirement *> Reqs = CommandGroup->getRequirements();
+  std::vector<detail::EventImplPtr> Events = CommandGroup->getEvents();
```
Comment (romanovvlad) [inline sycl/source/detail/scheduler/graph_builder.cpp]: Please, move this line closer to where it's used.

File: sycl/include/CL/sycl/detail/cg.hpp (hunk 1 above)
Comment (jbrodman) [reply sycl/include/CL/sycl/detail/cg.hpp]: clang format has now been run.

File: sycl/source/detail/scheduler/graph_builder.cpp (hunk 2 above)
Comment (jbrodman) [reply sycl/source/detail/scheduler/graph_builder.cpp]: Moving this line is probably going to cause more problems than it's worth.

The CG gets std::moved when the NewCmd is made.  I would either have to:
1) Not std::move it and deal with it later
2) Change the interface of ExecCGetcetc

**Pull request merged**